### PR TITLE
refactored codebase a/c napoleon style

### PR DESCRIPTION
**Pull request by RahulSankhla312**
- Changed the whole codebase Refactored codebase to consistently use Napoleon-style docstrings.
- Improves code readability and maintainability by providing clear function and class descriptions.

### CODE REVIEW

**Comment by RahulSankhla312** [comment]
@williballenthin Please review the PR and suggest me the required changes , if  necessary .

**Comment by mr-tz** [comment]
Neat! Please run this through black and follow https://google.github.io/styleguide/pyguide.html#381-docstrings.

**Comment by RahulSankhla312** [comment]
@mr-tz Please have a look . I have reformatted the codebase using black.

File: floss/api_hooks.py [hunk 1]
```diff
@@ -284,6 +367,13 @@ def __call__(self, emu, api, argv):
             s2 = fu.readStringAtRva(emu, s2va, maxsize=num)
 
             def cmp(a, b):
+                """
```
Comment (williballenthin) [inline floss/api_hooks.py:370]: please remove this comment

File: floss/api_hooks.py [hunk 2]
```diff
@@ -307,6 +401,10 @@ def __call__(self, emu, api, argv):
 
 
 class MemsetHook:
+    """
+    Hook calls to memset: fill memory with a constant byte.
```
Comment (williballenthin) [inline floss/api_hooks.py:405]: thank you for these comments, they are helpful

File: floss/decoding_manager.py [hunk 3]
```diff
@@ -59,34 +66,47 @@ def is_import(emu, va):
 
 @dataclass
 class Snapshot:
-    """
-    A snapshot of the state of the CPU and memory.
-
-    Attributes:
```
Comment (williballenthin) [inline floss/decoding_manager.py:65]: please restore these documentation comments

File: floss/api_hooks.py [hunk 4]
```diff
@@ -33,24 +33,35 @@
...
+        Log the start of an instruction.
+
+        """
         # overridden from Monitor
         # helpful for debugging decoders, but super verbose!
         logger.trace("prehook: 0x%x %s", startpc, op)
 
     def posthook(self, emu, op, endpc):
+        """
+
```
[the remark this answers is not in the document]
Comment (RahulSankhla312) [reply floss/api_hooks.py:61]: OK , I'll make the code consistent.

File: floss/api_hooks.py (hunk 2 above)
Comment (RahulSankhla312) [reply floss/api_hooks.py:405]: Thankyouu!!

File: floss/decoding_manager.py (hunk 3 above)
Comment (RahulSankhla312) [reply floss/decoding_manager.py:65]: OK sure .

**Comment by RahulSankhla312** [comment]
@williballenthin @mr-tz  I did the desired changes , please have a look.

File: floss/api_hooks.py [hunk 5]
```diff
@@ -205,42 +261,61 @@ def __call__(self, emu, api, argv):
 
 
 class CppNewObjectHook(MemoryAllocationHook):
-    """
-    Hook calls to:
+    """Hook calls to:
       - C++ new operator
     Thanks to @BenjaminSoelberg
+
```
[the remark this answers is not in the document]
Comment (RahulSankhla312) [reply floss/api_hooks.py]: I removed it .

**Comment by RahulSankhla312** [comment]
@williballenthin Is it ok now ?

**Comment by RahulSankhla312** [comment]
@williballenthin is all the changes ok for the PR to be merged ?

**Comment by mr-tz** [comment]
Please ensure all code style tests succeed and rebase to master.

**Pull request closed without merging**